src/components/TodoApp.tsx

```diff
@@ -0,0 +1,133 @@
+import { useContext, useEffect, useState } from 'react';
+
+import { TodosContext } from '../TodosContext';
+import TodoList from './TodoList';
+import TodosFilter from './TodosFilter';
+import { Todo } from '../types/Todo';
+import { Status } from '../enums/Status';
+
+const filteredTodos = (todos: Todo[], filter: Status): Todo[] => {
+  switch (filter) {
+    case Status.Active:
+      return todos.filter(todo => !todo.completed);
+    case Status.Completed:
+      return todos.filter(todo => todo.completed);
+    default:
+      return todos;
+  }
+};
+
+const TodoApp: React.FC = () => {
+  const { state, dispatch } = useContext(TodosContext);
+
+  const [title, setTitle] = useState('');
+  const [completedCount, setCompletedCount] = useState(0);
+  const [notCompletedCount, setNotCompletedCount] = useState(0);
+  const [isAllCompleted, setIsAllCompleted] = useState(false);
```

Comment:
You don't need states for these values, just check them dynamically, use variables